.github/workflows/tests_linux.yml

```diff
@@ -343,16 +343,38 @@ jobs:
           pip install --upgrade tensorflow~=$TF_VERSION keras~=$TF_VERSION
 
       - name: Install backend device
+        if: ${{ (((matrix.pl_backend != "all") }}
         run: |
           cd main
           CMAKE_ARGS="-DPL_BACKEND=${{ matrix.pl_backend }} -DCMAKE_PREFIX_PATH=${{ github.workspace }}/Kokkos -DENABLE_KOKKOS=ON -DENABLE_PYTHON=ON -DCMAKE_CXX_COMPILER=$(which g++-$GCC_VERSION)" \
           pip install -e . -vv
 
       - name: Run PennyLane-Lightning unit tests
+        if: ${{ (((matrix.pl_backend != "all") }}
         run: |
           cd main/
           DEVICENAME=`echo ${{ matrix.pl_backend }} | sed "s/_/./g"`
-          pytest tests/ $COVERAGE_FLAGS
+          python -m pytest tests/ $COVERAGE_FLAGS
+          pl-device-test --device ${DEVICENAME} --skip-ops --shots=20000 $COVERAGE_FLAGS --cov-append
+          pl-device-test --device ${DEVICENAME} --shots=None --skip-ops $COVERAGE_FLAGS --cov-append
+          mv coverage.xml coverage-${{ github.job }}.xml
+
+      - name: Install all backend devices
+        if: ${{ (((matrix.pl_backend == "all") }}
+        run: |
+          cd main
+          CMAKE_ARGS="-DPL_BACKEND="lightning_qubit" -DCMAKE_PREFIX_PATH=${{ github.workspace }}/Kokkos -DENABLE_KOKKOS=ON -DENABLE_PYTHON=ON -DCMAKE_CXX_COMPILER=$(which g++-$GCC_VERSION)" \
+          pip install -e . -vv
+          rm -rf build
+          CMAKE_ARGS="-DPL_BACKEND="lightning_kokkos" -DCMAKE_PREFIX_PATH=${{ github.workspace }}/Kokkos -DENABLE_KOKKOS=ON -DENABLE_PYTHON=ON -DCMAKE_CXX_COMPILER=$(which g++-$GCC_VERSION)" \
+          pip install -e . -vv
+
+      - name: Run PennyLane-Lightning unit tests for lightning.qubit with all devices installed
+        if: ${{ (((matrix.pl_backend == "all") }}
+        run: |
+          cd main/
+          DEVICENAME= "lightning.qubit"
+          python -m pytest tests/ $COVERAGE_FLAGS
```

Comment:
I don't see this as a lot of if-else. An alternative to the if-else`s here would be to have a workflow exclusive to all devices, and this would imply lots of code duplication. Let me know your thoughts about that.

Reply:
Ok, let's leave as is for now.

Reply:
We could maybe allow `lightning_all` in `setup.py` and merge this block with the one above. Or leave it for now.